### PR TITLE
Added diagnostics to all errors

### PR DESCRIPTION
**Pull request by arthurfiorette**
This PR standardizes all errors thrown by this library, which helps every package that uses this code underneath, as well with the end-user using this CLI.

As every errors extends from `TJSGError`, every error may have a `.diagnostic` which can be formatted by the `ts.formatDiagnosticsWithColorAndContext` function.

For example, when having a typecheck error, this is what was shown:

image

VS what is now:

image

Or, for example, when a node cannot be parsed (I faked the error):

image

VS what is now:

image

---

All errors are serializable and can contain useful information. 

In non `tty` environment, such as CI environments, thrown errors doesn't emit ANSI color codes or anything might blowup logs with useless information.

image

# Version

Published prerelease version: `v2.3.0-next.0`

<details>
  <summary>Changelog</summary>

  #### 🚀 Enhancement
  
  - Added diagnostics to all errors [#1963](https://github.com/vega/ts-json-schema-generator/pull/1963) ([@arthurfiorette](https://github.com/arthurfiorette))
  
  #### Authors: 1
  
  - Arthur Fiorette ([@arthurfiorette](https://github.com/arthurfiorette))
</details>

<details>
  <summary>📦 Published PR as canary version: <code>2.3.0--canary.1963.49df2de.0</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install ts-json-schema-generator@2.3.0--canary.1963.49df2de.0
  # or 
  yarn add ts-json-schema-generator@2.3.0--canary.1963.49df2de.0
  ```
</details>

### CODE REVIEW

File: src/Error/Errors.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
+import ts from "typescript";
+import { PartialDiagnostic, TJSGError } from "./BaseError.js";
+import type { BaseType } from "../Type/BaseType.js";
+import { JSONSchema7 } from "json-schema";
+
+export class UnknownNodeTJSGError extends TJSGError {
```
Comment (domoritz) [inline src/Error/Errors.ts]: I think I'd prefer simpler names like `UnknownNodeError `

**Review by domoritz — APPROVED** [review]

**Comment by arthurfiorette** [comment]
Next up, I was thinking of rewriting the CLI to use [oclif](https://oclif.io/). I did the same thing with kitajs and it's awesome. It auto generates cli documentation, provides autocompletions, automatically creates help commands, detect cli updates and much more automatically.

Sample of kitajs cli autocomplete:

image

Sample of kitajs cli help:

image

What do you think?

File: src/Error/Errors.ts (hunk 1 above)
Comment (arthurfiorette) [reply src/Error/Errors.ts]: sure!

**Comment by domoritz** [comment]
Better cli sounds great

**Comment by domoritz** [comment]
I gave you write permissions to the repo so you can make PRs from internal branches and merge PRs when they are done.

**Comment by arthurfiorette** [comment]
thanks!

**Pull request merged**